compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala

```diff
@@ -352,8 +352,14 @@ class ReifyQuotes extends MacroTransform {
             transformSplice(spliceTree)
 
           case tree: TypeTree if tree.tpe.typeSymbol.isSplice =>
-            val splicedType = tree.tpe.stripTypeVar.asInstanceOf[TypeRef].prefix.termSymbol
-            transformSplice(ref(splicedType).select(tpnme.splice).withSpan(tree.span))
+            tree.tpe.stripTypeVar match {
+              case tp: AppliedType =>
+                val splicedType = tp.tycon.asInstanceOf[TypeRef].prefix.termSymbol
+                AppliedTypeTree(transformSplice(ref(splicedType).select(tpnme.splice)), tp.args.map(TypeTree(_))).withSpan(tree.span)
```

Comment:
It is not obvious why there is no need to `transformSplice` on the type arguments.

Reply:
Maybe I should call `transform` on the arguments. I will double check.

Reply:
Actually, this whole branch is useless. This is already handled by the transformation done in the `Staging` phase.